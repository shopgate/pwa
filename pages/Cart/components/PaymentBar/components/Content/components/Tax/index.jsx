import React from 'react';
import PropTypes from 'prop-types';
import CartTotalLine from '@shopgate/pwa-ui-shared/CartTotalLine';
import CartContext from 'Pages/Cart/context';
import connect from './connector';

/**
 * The Tax component.
 * @returns {JSX}
 */
const Tax = ({ taxData }) => {
  if (!taxData) {
    return null;
  }

  const { label, amount } = taxData;
  return (
    <CartContext.Consumer>
      {({ currency, isLoading }) => (
        <CartTotalLine isDisabled={isLoading} type="tax">
          <CartTotalLine.Label label={label} />
          <CartTotalLine.Amount amount={amount} currency={currency} />
        </CartTotalLine>
      )}
    </CartContext.Consumer>
  );
};

Tax.propTypes = {
  taxData: PropTypes.shape({
    label: PropTypes.string,
    amount: PropTypes.number,
  }),
};

Tax.defaultProps = {
  taxData: null,
};

export default connect(Tax);
