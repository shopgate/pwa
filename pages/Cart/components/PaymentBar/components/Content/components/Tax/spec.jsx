import React from 'react';
import { shallow } from 'enzyme';
import Tax from './';

jest.mock('Pages/Cart/context', () => ({
  Consumer: jest.fn(({ children }) => children({
    currency: 'EUR',
    config: {
      hideTax: false,
      tax: {
        text: null,
        hint: null,
      },
    },
  })),
}));
jest.mock('./connector', () => cmp => cmp);

describe('<Tax />', () => {
  it('should render and match snapshot', () => {
    // eslint-disable-next-line extra-rules/no-single-line-objects
    const wrapper = shallow(<Tax taxData={{ label: 'Tax', amount: 10 }} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
  it('should not render and match snapshot', () => {
    const wrapper = shallow(<Tax taxData={null} />);
    expect(wrapper).toBeEmptyRender();
  });
});
