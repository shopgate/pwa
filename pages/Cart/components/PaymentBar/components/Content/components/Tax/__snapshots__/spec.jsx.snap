// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tax /> should render and match snapshot 1`] = `
<CartTotalLine
  isDisabled={false}
  type="tax"
>
  <Label
    label="Tax"
    showSeparator={true}
  />
  <Amount
    amount={10}
    currency="EUR"
  />
</CartTotalLine>
`;
